--- origsrc/apache-log4cxx-0.10.0/src/main/cpp/odbcappender.cpp	2008-03-31 17:34:09.000000000 -0500
+++ src/apache-log4cxx-0.10.0/src/main/cpp/odbcappender.cpp	2009-01-14 23:53:05.499625000 -0600
@@ -27,7 +27,7 @@
 #endif
 #include <log4cxx/private/log4cxx_private.h>
 #if LOG4CXX_HAVE_ODBC
-#if defined(WIN32) || defined(_WIN32)
+#if defined(WIN32) || defined(_WIN32) || defined(__CYGWIN__)
 #include <windows.h>
 #endif
 #include <sqlext.h>
@@ -167,8 +167,13 @@ void ODBCAppender::execute(const LogStri
          throw SQLException( SQL_HANDLE_DBC, con, "Failed to allocate sql handle.", p);
       }
 
+#if LOG4CXX_WCHAR_T_API
       SQLWCHAR* wsql = Transcoder::wencode(sql, p); 
       ret = SQLExecDirectW(stmt, wsql, SQL_NTS);
+#else
+      SQLCHAR* nsql = reinterpret_cast<SQLCHAR*>(Transcoder::encode(sql, p)); 
+      ret = SQLExecDirect(stmt, nsql, SQL_NTS);
+#endif
 
      if (ret < 0)
       {
@@ -237,6 +242,7 @@ ODBCAppender::SQLHDBC ODBCAppender::getC
       }
 
 
+#if LOG4CXX_WCHAR_T_API
      SQLWCHAR* wURL = Transcoder::wencode(databaseURL, p);
 
      wchar_t szOutConnectionString[1024];
@@ -246,7 +252,17 @@ ODBCAppender::SQLHDBC ODBCAppender::getC
             wURL, SQL_NTS, 
             szOutConnectionString, sizeof( szOutConnectionString ),
             &nOutConnctionLength, SQL_DRIVER_NOPROMPT );
+#else
+     SQLCHAR* nURL = reinterpret_cast<SQLCHAR*>(Transcoder::encodeUTF8(databaseURL, p));
+
+     SQLCHAR szOutConnectionString[1024];
+     SQLSMALLINT nOutConnctionLength = 0;
 
+     ret = SQLDriverConnect( connection, NULL, 
+            nURL, SQL_NTS, 
+            szOutConnectionString, sizeof( szOutConnectionString ),
+            &nOutConnctionLength, SQL_DRIVER_NOPROMPT );
+#endif
 
      if (ret < 0)
       {
